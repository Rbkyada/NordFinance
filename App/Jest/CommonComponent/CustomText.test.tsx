import renderer from 'react-test-renderer';
import { Text } from 'react-native';
import { CustomText } from '@CommonComponent/CustomText';

describe('CustomText', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CustomText>Test Text</CustomText>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with custom style', () => {
    const customStyle = { fontWeight: 'bold', fontSize: 16 };
    const component = renderer.create(
      <CustomText style={customStyle}>Custom Style Text</CustomText>,
    );
    const text = component.root.findByType(Text);
    expect(text.props.style).toContainEqual(customStyle);
  });

  it('renders with specified font size', () => {
    const component = renderer.create(
      <CustomText size={20}>Custom Font Size</CustomText>,
    );
    const text = component.root.findByType(Text);
    expect(text.props.style).toContainEqual({ fontSize: 20 });
  });

  it('triggers onPress event when the text is pressed', () => {
    const onPressMock = jest.fn();
    const component = renderer.create(
      <CustomText onPress={onPressMock}>Clickable Text</CustomText>,
    );
    const text = component.root.findByType(Text);
    text.props.onPress();
    expect(onPressMock).toHaveBeenCalled();
  });
});
