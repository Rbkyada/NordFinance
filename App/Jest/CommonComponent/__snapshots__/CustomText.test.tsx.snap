// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomText renders correctly 1`] = `
<Text
  numberOfLines={0}
  style={
    [
      {
        "fontSize": 14,
      },
      {
        "color": "#1c1c1e",
      },
      {
        "fontFamily": "DMSans-Regular",
      },
      undefined,
    ]
  }
>
  Test Text
</Text>
`;
